### ISSUE TITLE
handling scripts

### ISSUE THREAD

**Issue by JensWalter** [apimeister/unpatched-server#25]
current impl
![Screenshot 2023-08-16 at 11 52 06](https://github.com/apimeister/unpatched-server/assets/1734701/cda536cd-f295-422c-b99e-40e7a585e71e)

we should refactor to the following interaction pattern
POST /script -> create new script, id is returned but not posted, version can be posted but is optional with default 0.0.1
GET /script -> retrieve all as is
GET /script/{id} -> retrieve one script, most up to date version
GET /script/{id}/{version} -> retrieve one script with specific version
POST /script/{id} -> update one, but we need to inform the server about the version increase, patch,minor,major, server then return new version in response
DELETE /script/{id} -> delete all versions of a script

**Comment by JensWalter** [comment]
if the id identifies the script instance, we need another criteria to determine the script branch (all versions), name is not gonna help, because this will collide heavily.

**Comment by alexkunde** [comment]
this would mean scripts with same names and versions would be living side by side. If we look into the direction of Crates.io, NPM or other packagemanagers the name seems to be the unique identifier. If we look into github, repo/name would be the identifier. In our case, we could thinking script collections? This would open up the naming without losing identifiability.

**Comment by JensWalter** [comment]
yes, but those names are scoped by user credentials, so far this scope does not exist.
